appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/jdbc/JdbcPostgresTest.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.appsignals.test.jdbc;
+
+import static io.opentelemetry.proto.trace.v1.Span.SpanKind.SPAN_KIND_CLIENT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertThat;
+
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.metrics.v1.ExponentialHistogramDataPoint;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.PullPolicy;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.lifecycle.Startable;
+import software.amazon.opentelemetry.appsignals.test.base.ContractTestBase;
+import software.amazon.opentelemetry.appsignals.test.utils.AppSignalsConstants;
+import software.amazon.opentelemetry.appsignals.test.utils.ResourceScopeMetric;
+import software.amazon.opentelemetry.appsignals.test.utils.ResourceScopeSpan;
+import software.amazon.opentelemetry.appsignals.test.utils.SemanticConventionsConstants;
+
+@Testcontainers(disabledWithoutDocker = true)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class JdbcPostgresTest extends ContractTestBase {
+
+  private static final String DB_SYSTEM = "postgresql";
+  private static final String DB_NAME = "testdb";
+  private static final String DB_USER = "sa";
+  private static final String DB_PASSWORD = "password";
+  private static final String DB_OPERATION = "SELECT";
+
+  private static final String NETWORK_ALIAS = "postgres";
+
+  private PostgreSQLContainer<?> postgreSqlContainer;
+
+  @AfterEach
+  public void afterEach() {
```

Comment:
The application [creates a table on startup](https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/appsignals-tests/images/jdbc/src/main/java/software/amazon/opentelemetry/AppController.java#L44) and the application [is restarted after each test](https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/base/ContractTestBase.java#L119).

The problem is that the postgres container is not restarted which means that all the tests after the first one won't be able to create any table. Also we should reset all the state between tests so that we prevent one test from breaking another.

In the previous existent test we were using H2 which lives inside the application process, so if the application restarts, the database will also be cleared.

Reply:
Clarified offline: `postgreSqlContainer.stop(); stops the container, but when we start the application again for the second test, it will start the dependencies again(in this case postgres) because it is not running.